code/memcheck/CMakeLists.txt

```
@@ -5,23 +5,27 @@ project( memcheck LANGUAGES CXX )
 
 # Set up the compilation environment.
 include( "${CMAKE_CURRENT_SOURCE_DIR}/../CompilerSettings.cmake" )
+include( "${CMAKE_CURRENT_SOURCE_DIR}/../SolutionTarget.cmake" )
 
 # Create the user's library.
-add_library( memcheckPoly Polygons.hpp Polygons.cpp )
+add_library( memcheckPoly "Polygons.hpp" "Polygons.cpp" )
 set_target_properties( memcheckPoly PROPERTIES OUTPUT_NAME "poly" )
+if( "${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC" )
```

cmake supports this directly:
```suggestion
if(MSVC)
```
appears in other places too.

---

Interestingly enough the [MSVC](https://cmake.org/cmake/help/latest/variable/MSVC.html) variable is still considered a recommended variable to use, though I suspect it's because making it deprecated would upset too many people. 🤔 

Its "GCC equivalent" on the other hand, [CMAKE_COMPILER_IS_GNUCXX](https://cmake.org/cmake/help/latest/variable/CMAKE_COMPILER_IS_GNUCXX.html), has been deprecated since a while.

Long story short, in any new code that I write I try to only use [CMAKE_CXX_COMPILER_ID](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html) to decide what to do for different compilers. As it generally makes it a lot easier to extend the code for additional compilers down the line.

---

Fine for me!